### ISSUE TITLE
Constant Tracking Is Order Dependent

### ISSUE THREAD

**Issue by sebmarkbage** [babel/babel#2550]
This results in a constant binding:

```js
bar = 2;
var bar = 1;
```

So does this:

```js
function foo() {
  bar = 2;
}
var bar = 1;
```

According to `scope.bindings.bar.constant`, but I would expect it to be `false`.

(Babel 5.8.25)

**Comment by sebmck** [comment]
Duplicate of #1942. This was caused by the scope tracking being consolidated into a single pass rather than separate ones for each scope and "reference collection".

**Comment by sebmck** [comment]
This has since been backported and fixed in 5.8.28.

**Comment by sebmarkbage** [comment]
<3 You're the best! I was worried how I was going to work around this.

So good that you're not just dropping the old branch while working on the new! That can be really hurtful to projects and one reason React choses to do things incrementally.
